src/NuGet.Core/NuGet.PackageManagement/BuildIntegration/DependencyGraphRestoreUtility.cs

```diff
@@ -370,4 +404,13 @@ public static async Task<PackageSpec> GetProjectSpec(IDependencyGraphProject pro
             return restoreContext;
         }
     }
+
+    internal class ProjectRestoreSpec
```

Comment:
There should only be 1 class per file, with the exception of nested classes. If we can't get rid of this class as per my other comment above, then this class definition should be moved inside `DependencyGraphRestoreUtility`, and made `private`, not `internal`, since it's not used outside `DependencyGraphRestoreUtility`.

Reply:
Ok. I moved ProjectRestoreSpec class inside DependencyGraphRestoreUtility class and made it private.